subprojects/core/src/main/java/org/gradle/execution/taskgraph/DefaultTaskGraphExecuter.java

```diff
@@ -225,31 +227,36 @@ private void ensurePopulated() {
      */
     private class EventFiringTaskWorker implements Action<TaskInternal> {
         private final TaskExecuter taskExecuter;
-        private final BuildOperationExecutor.Operation parentOperation;
+        private final BuildOperationState parentOperation;
 
-        EventFiringTaskWorker(TaskExecuter taskExecuter, BuildOperationExecutor.Operation parentOperation) {
+        EventFiringTaskWorker(TaskExecuter taskExecuter, BuildOperationState parentOperation) {
             this.taskExecuter = taskExecuter;
             this.parentOperation = parentOperation;
         }
 
         @Override
         public void execute(final TaskInternal task) {
-            TaskOperationDescriptor taskOperation = new TaskOperationDescriptor(task);
-            BuildOperationDetails buildOperationDetails = BuildOperationDetails.displayName("Task " + task.getIdentityPath()).name(task.getIdentityPath().toString()).parent(parentOperation).operationDescriptor(taskOperation).build();
-            buildOperationExecutor.run(buildOperationDetails, new Action<BuildOperationContext>() {
+            buildOperationExecutor.setRootOperationOfCurrentThread(parentOperation);
```

Comment:
Yes. I was thinking in another direction. But you are right. I changed that back. We now only have `getCurrentOperationId()` and `getCurrentParentOperationId()` on the executor. I am not sure if we need the latter on the long run.

Exposing only the ID (and no further state) led to some more changes in the executor implementation. The main issue was to determine the runtime state of the parent for the _is parent running_ check. I ended up with maintaining a set of running operation ids. I think this is the cheapest way to handle this.